src/components/Card.js

```diff
@@ -26,26 +30,41 @@ class Card {
 
     this._likeButton = this._element.querySelector(".card__like-button");
     this._deleteButton = this._element.querySelector(".card__delete-button");
-    this._likeButton.addEventListener("click", () => this.togglelikeButton());
-    this._deleteButton.addEventListener("click", () => this.remove());
+    this._likeButton.addEventListener("click", () => this.toggleLikeButton());
+    this._deleteButton.addEventListener("click", () => {this._handleDeleteClick(this)});
   }
 
-  togglelikeButton() {
-    this._likeButton.classList.toggle("card__like-button_active");
-  }
+  toggleLikeButton() {
+    const isLiked = this._likeButton.classList.contains("card__like-button_active");
+ this._handleUserLikes(this._cardId, !isLiked)
+     .then((updatedCard) => {
+         this._likes = updatedCard.likes;
+         this._likeButton.classList.toggle("card__like-button_active");
+    })
+    .catch((err) => console.log("Error updating like status:", err))};
 
-  remove() {
+  deleteCard(){
     this._element.remove();
   }
 
+  getId(){
+    return this._cardId;
+  }
+
   getView() {
     this._element = this._getTemplate();
     this._element.querySelector(".card__image").src = this._link;
     this._element.querySelector(".card__image").alt = this._name;
     this._element.querySelector(".card__title").textContent = this._name;
     this._setEventListeners();
+    const hasUserLiked = this._likes.some((user) => user._id === this._userId);
+    if (this._likes.some((like) => like._id === this._userId)) {
```

Comment:
* When I reload the page, all **black** hearts become **white** again.

`    this._likes = cardData.likes || [];
`

`cardData.likes` doesn't exist


You should define `this._isLiked`


```js

    this._isLiked = cardData.isLiked

```


So, you'll check it like this


```js
    if (this._isLiked) {
      this._likeButton.classList.add("card__like-button_active");
    }

```

Reply:
But something is wrong with the `isLiked` key. Now it doesn't exist in the card data either...


Ok, I'll accept it as it is, because my variant doesn't work as well